src/__stories__/components/Colors.tsx

```diff
@@ -18,19 +18,18 @@ const Colors = () => {
           </Typography>
           <div style={{ display: 'flex', gap: 24, marginTop: 16 }}>
             <div style={{ display: 'flex', flexDirection: 'column', gap: 16 }}>
-              {Object.keys(
-                (theme.colors as Record<string, Record<string, string>>)[
-                  sentiment
-                ],
-              )
+              {Object.keys(theme.colors[sentiment as keyof typeof theme.colors])
```

Comment:
Isn't this useless ? Typescript is not able to infer sentiment type ?

Reply:
Alright, to avoid overwriting types then :)


```tsx
import lightTheme, { Color } from '../../theme'
import ThemeWrapper from './ThemeWrapper'

type AvailableContexts = keyof typeof lightTheme['colors'][Color]

const Colors = () => {
  const theme = useTheme()

  return (
    <>
      {Object.keys(theme.colors).map((sentiment) => (
        <>
          <Typography variant="title" style={{ textTransform: 'capitalize' }}>
            {sentiment}
          </Typography>
          <div style={{ display: 'flex', gap: 24, marginTop: 16 }}>
            <div style={{ display: 'flex', flexDirection: 'column', gap: 16 }}>
              {Object.keys(theme.colors[sentiment as Color])
                .filter(context => context.includes('background'))
                .map(context => (
                  <BorderedBox
                    key={context}
                    style={{
                      alignItems: 'center',
                      backgroundColor: theme.colors[sentiment as Color][context as AvailableContexts],
```

Reply:
Alright, to avoid overwriting types then :)


```tsx
import lightTheme, { Color } from '../../theme'
import ThemeWrapper from './ThemeWrapper'


type ColorContext = keyof typeof lightTheme['colors'][Color]

const Colors = () => {
  const theme = useTheme()

  return (
    <>
      {(Object.keys(theme.colors) as Color[]).map(sentiment => {
        const colorContextKeys = Object.keys(
          theme.colors[sentiment],
        ) as ColorContext[]

        return (
          <>
            <Typography variant="title" style={{ textTransform: 'capitalize' }}>
              {sentiment}
            </Typography>
            <div style={{ display: 'flex', gap: 24, marginTop: 16 }}>
              <div
                style={{ display: 'flex', flexDirection: 'column', gap: 16 }}
              >
                {colorContextKeys
                  .filter(context => context.includes('background'))
                  .map(context => (
                    <BorderedBox
                      key={context}
                      style={{
                        alignItems: 'center',
                        backgroundColor: theme.colors[sentiment][context],
```